presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java

```diff
@@ -31,30 +31,30 @@ public ExtendedFileSystem createCachingFileSystem(
             ExtendedFileSystem fileSystem,
             CacheManager cacheManager,
             boolean cachingEnabled,
-            CacheType cacheType,
-            boolean validationEnabled)
+            CacheConfig cacheConfig)
             throws IOException
     {
         if (!cachingEnabled) {
             return fileSystem;
         }
 
-        checkState(cacheType != null);
+        checkState(cacheConfig.getCacheType() != null);
```

Comment:
nit: Do you also want to check whether `cacheConfig` is null or not?

Reply:
good catch,  just fixed it